shuffle/cli/tests/shuffle-forge-cli-tests.rs

```diff
@@ -0,0 +1,171 @@
+
+impl Test for TransactionsWithoutNetworksToml {
+    fn name(&self) -> &'static str {
+        "shuffle::transactions-without-networks-toml"
+    }
+}
+
+impl AdminTest for TransactionsWithoutNetworksToml {
+    fn run<'t>(&self, _ctx: &mut AdminContext<'t>) -> Result<()> {
+        let temp_dir = tempdir()?;
+        let base_path = temp_dir.path();
+        let base_path_string = base_path.to_string_lossy().to_string();
+        let output = std::process::Command::new(BINARY)
+            .args([
+                HOME_PATH,
+                base_path_string.as_str(),
+                "transactions",
+                "--network",
+                "forge",
+            ])
+            .output()?;
+        let std_err = String::from_utf8(output.stderr)?;
+        assert_eq!(
+            std_err
+                .contains("Error: Please add specified network to the ~/.shuffle/Networks.json\n"),
+            true
+        );
+        assert_eq!(output.status, ExitStatus::from_raw(256));
+        Ok(())
+    }
+}
+
+pub struct TransactionsWithoutAccount;
+
+impl Test for TransactionsWithoutAccount {
+    fn name(&self) -> &'static str {
+        "shuffle::transactions-without-account"
+    }
+}
+
+impl AdminTest for TransactionsWithoutAccount {
+    fn run<'t>(&self, ctx: &mut AdminContext<'t>) -> Result<()> {
+        let temp_dir = tempdir()?;
+        let base_path = temp_dir.path();
+        let base_path_string = base_path.to_string_lossy().to_string();
+
+        let helper = ShuffleTestHelper::new(ctx.chain_info(), base_path)?;
+        let client = ctx.client();
+        let factory = ctx.chain_info().transaction_factory();
+        enable_open_publishing(&client, &factory, ctx.chain_info().root_account())?;
+        helper.create_project()?;
+        helper.overwrite_networks_config_into_toml(helper.networks_config())?;
+        let output = std::process::Command::new(BINARY)
+            .args([
+                HOME_PATH,
+                base_path_string.as_str(),
+                "transactions",
+                "--raw",
+                "--network",
+                "forge",
+            ])
+            .output()?;
+        let std_err = String::from_utf8(output.stderr)?;
+
+        assert_eq!(
+            std_err
+                .contains("Error: An account hasn't been created yet! Run shuffle account first\n"),
+            true
+        );
+        assert_eq!(output.status, ExitStatus::from_raw(256));
+        Ok(())
+    }
+}
+
+pub struct TransactionsWithNetworkRawAddressFlags;
+
+impl Test for TransactionsWithNetworkRawAddressFlags {
+    fn name(&self) -> &'static str {
+        "shuffle::transactions-with-networks-raw-and-address-flags"
+    }
+}
+
+impl AdminTest for TransactionsWithNetworkRawAddressFlags {
+    fn run<'t>(&self, ctx: &mut AdminContext<'t>) -> Result<()> {
+        let temp_dir = tempdir()?;
+        let base_path = temp_dir.path();
+        let helper = bootstrap_shuffle(ctx, base_path)?;
```

Comment:
Can we keep the creation of the `tempdir()` inside `ShuffleTestHelper`? Instead, add a getter `helper.home_path()` or wtever you need to pass into the CLI.